File: e71d1c2a-be33-4ffb-916a-fa2be47b15b6/src/main/java/com/court/execution/repository/UserRepository.java
package com.court.execution.repository;

import com.court.execution.entity.User;
import com.court.execution.entity.UserRole;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByUsername(String username);

    boolean existsByUsername(String username);

    List<User> findByRole(UserRole role);

    List<User> findByEnabledTrue();
}

File: 9096d8e2-d12b-468a-91ae-53c97e7bf587/backend/src/main/kotlin/com/ems/dispatch/repository/UserRepository.kt
package com.ems.dispatch.repository

import com.ems.dispatch.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByUsername(username: String): User?
    fun findByRolesContaining(role: String): List<User>
    fun existsByUsername(username: String): Boolean
}
